examples/step-93/step-93.cc

```diff
@@ -0,0 +1,840 @@
+  // solver. For smaller problems, one can also use a direct solver
+  // (see step-29) for which you would just replace the main part of
+  // this function by the following three lines of code:
+  //
+  //   `SparseDirectUMFPACK direct_solver;`
+  //
+  //   `direct_solver.initialize(system_matrix);`
+  //
+  //   `direct_solver.vmult(solution, system_rhs);`
+  template <int dim>
+  void Step93<dim>::solve()
+  {
+    std::cout << "Beginning solve..." << std::endl;
+    Timer timer;
+
+    SolverControl solver_control(1'000'000, 1e-6 * system_rhs.l2_norm());
+    SolverMinRes<Vector<double>> solver(solver_control);
+
+    solver.solve(system_matrix, solution, system_rhs, PreconditionIdentity());
+
+    timer.stop();
+    std::cout << "Wall time: " << timer.wall_time() << "s" << std::endl;
+    std::cout << "Solved in " << solver_control.last_step()
+              << " MINRES iterations." << std::endl;
+  }
+
+
+  // @sect4{Step93::output_results()}
+
+  // The `output_results()` function is a bit more robust for this program than
+  // is typical. This is because, in order to visualize the heat sources we have
+  // optimized, we need to do extra work and interpolate them onto a mesh. We do
+  // this by instantiating a new DoFHandler object and then using the helper
+  // function VectorTools::interpolate().
+  //
+  // The top of the function is as always when using vector-valued
+  // elements (see, for example, step-22) and simply outputs all of
+  // the solution variables on the mesh cells they are defined on:
+  template <int dim>
+  void Step93<dim>::output_results() const
+  {
+    const std::vector<std::string> solution_names = {"u", "lambda", "c"};
+
+    const std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      interpretation = {DataComponentInterpretation::component_is_scalar,
+                        DataComponentInterpretation::component_is_scalar,
+                        DataComponentInterpretation::component_is_scalar};
+
+    DataOut<dim> data_out;
+    data_out.add_data_vector(dof_handler,
+                             solution,
+                             solution_names,
+                             interpretation);
+
+    // The non-local degrees of freedom are of course defined on the
+    // first several cells, but at least logically are considered to
+    // live everywhere (or, if you prefer, nowhere at all since they
+    // do not represent spatial functions). But, conceptually, we use
+    // them as multipliers for the heat sources, and so while the
+    // coefficients $C^k$ are non-local, the heat source $\sum_k C^k
+    // f_k(\mathbf x)$ *is* a spatially variable function. It would be
+    // nice if we could visualize that as well. The same is true for
+    // the target heat distribution $\bar u$ we are trying to match.
+    //
+    // To do so, we
+    // create a new dummy dof handler to output the target function
```

Comment:
Like above - I believe avoiding words like 'dummy' and 'toy' and instead using more ad hoc descriptions makes this easier to understand.